swift/apple/FirezoneKit/Sources/FirezoneKit/Keychain/Keychain.swift

```diff
@@ -43,95 +42,84 @@ public actor Keychain {
   public init() {}
 
   public func add(query: [CFString: Any]) async throws {
-    return try await withCheckedThrowingContinuation { [weak self] continuation in
-      self?.workQueue.async {
-        var ref: CFTypeRef?
-        let ret = SecStatus(SecItemAdd(query as CFDictionary, &ref))
-        guard ret.isSuccess else {
-          continuation.resume(
-            throwing: KeychainError.appleSecError(call: "SecItemAdd", status: ret))
-          return
-        }
-
-        continuation.resume()
+    return try await withCheckedThrowingContinuation { continuation in
+      var ref: CFTypeRef?
+      let ret = SecStatus(SecItemAdd(query as CFDictionary, &ref))
```

Comment:
Is this really so "blocking" that it needs to be in a continuation? Maybe we can just run it as a blocking call within the current async function?

Reply:
I'll investigate. These calls involve the secure enclave and disk I/O so we probably want to maintain some mechanism for async.

Reply:
Ok so TIL: Actors (which this Keychain class is) are used to enforce concurrency calling from other threads. I didn't write the original implementation of this so I suspect the decision here was that we always want calls to the Keychain to happen in the order they are called.

Outsiders that call an actor's functions need to use `await` because of this.

Nothing I can find in Apple's documentation or via resources on the web mention using an actor as a best practice for working with the keychain, so I this was simply a case of overengineering.

I'll make these functions static which will clean things up nicely.

Reply:
It's going to be a bit of refactoring. I'll open a new PR.